security/pkg/caclient/client.go

```diff
@@ -27,51 +27,35 @@ import (
 
 // CAClient is a client to provision key and certificate from the upstream CA via CSR protocol.
 type CAClient struct {
-	platformClient platform.Client
-	protocolClient grpc.CAGrpcClient
-	istioCAAddress string
-
-	identity    string
-	identityOrg string
-	rSAKeySize  int
-	ttl         time.Duration
```

Comment:
I think we still need these fields... How does node agent get the first key/cert for a new workload, if we don't have these specs?

Reply:
In management/server.go, WorkloadAdded, the identity is retrieved from request.Attrs.ServiceAccount, the Org and RSAKeySize are part of CAClientConfig (a field of NodeAgent server). The value will be passed from cmd flag.